src/app/indexed-db/db-changes.service.ts

```diff
@@ -154,19 +169,25 @@ export class DbChangesService {
   }
 
   async deleteNonEnergyBenefit(nonEnergyBenefit: IdbNonEnergyBenefit) {
+    //delete kpm impacts
+    let keyPerformanceMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.keyPerformanceMetricImpactsIdbService.keyPerformanceMetricImpacts.getValue();
+    let nebKpmImpacts: Array<IdbKeyPerformanceMetricImpact> = keyPerformanceMetricImpacts.filter(metricImpact => { return metricImpact.nebId == nonEnergyBenefit.guid; })
+    await this.deleteKeyPerformanceMetricImpacts(nebKpmImpacts);
+
     await firstValueFrom(this.nonEnergyBenefitsIdbService.deleteWithObservable(nonEnergyBenefit.id));
     await this.nonEnergyBenefitsIdbService.setNonEnergyBenefits();
   }
 
   async deleteEnergyOpportunity(energyOpportunity: IdbEnergyOpportunity) {
```

Comment:
may consider updating the `deleteEnergyOpportunities()` method below with this `deleteEnergyOpportunity()`